frontend/src/components/NavigationContainer/NavigationBar/UserButtons/UserDropDown/index.tsx

```diff
@@ -54,14 +56,41 @@ export const UserDropDown = () => {
                 }`}
             >
                 {/** Container */}
-                <div className="py-2 bg-surface-2 rounded">
-                    <div className=" h-12 w-full">
-                        <div
-                            className="w-full h-full px-3 flex flex-row gap-1 justify-start items-center bg-neutral-10 bg-opacity-0 hover:bg-opacity-5 active:bg-opacity-8 text-lg"
-                            onClick={logout}
-                        >
-                            {solidIcon('LogoutIcon')}
-                            Log out
+                <div className="flex flex-col py-2 bg-surface-2 rounded">
+                    {/** User name */}
+                    <div className="h-10 w-full">
+                        <div className="w-full h-full px-3 text-lg flex items-center">
+                            Username
+                        </div>
+                    </div>
+
+                    {/** Divider */}
+                    <div className="py-3 px-2">
+                        <div className="h-px w-full bg-neutral-10"></div>
+                    </div>
+
+                    {/** Actions */}
+                    <div className="flex flex-col gap-2 cursor-default">
+                        {/** Settings */}
+                        <div className="h-10 w-full ">
+                            <div
+                                className="w-full h-full px-3 flex flex-row gap-2 justify-start items-center bg-neutral-10 bg-opacity-0 hover:bg-opacity-5 active:bg-opacity-8 text-lg"
+                                onClick={() => navigate('/settings')}
+                            >
+                                {solidIcon('SettingsIcon')}
+                                Settings
+                            </div>
+                        </div>
+
+                        {/** Logout */}
+                        <div className=" h-10 w-full">
+                            <div
+                                className="w-full h-full px-3 flex flex-row gap-2 justify-start items-center bg-neutral-10 bg-opacity-0 hover:bg-opacity-5 active:bg-opacity-8 text-lg text-red"
+                                onClick={logout}
+                            >
+                                {solidIcon('LogoutIcon')}
+                                Log out
+                            </div>
```

Comment:
Would it be possible to split this into smaller components?

Reply:
Totally if wanted.
I just needed 2 similar components thus I only copypasted the old code.
I'll just leave it now like this and If more buttons needed in the future, then new component can be created for it.